tests/integration/basic/test_analyze_dataset.py

```diff
@@ -16,11 +16,11 @@
 
 
 class TestInferTypes(unittest.TestCase):
-    def test_infer_types_on_home_rentlas(self):
+    def test_analyze_home_rentlas(self):
         datasource = FileDS(
             "https://raw.githubusercontent.com/mindsdb/mindsdb-examples/master/classics/home_rentals/dataset/train.csv"
         )
-        type_information = infer_types(datasource.df, pct_invalid=0)
+        type_information = analyze_dataset(datasource.df).type_information
```

Comment:
how is `analyze_dataset` diff from `infer_types` now?

Reply:
`analyze_dataset` = > return a data analysis class that contains both the type information and the stat analysis